libs/types/src/cdf/election-results-reporting/convert.ts

```diff
@@ -0,0 +1,269 @@
+import { assertDefined, err, ok, Result } from '@votingworks/basics';
+import getDeepValue from 'lodash.get';
+import { Candidate, CandidateId } from '../../election';
+import { LanguageCode } from '../../language_code';
+import * as ResultsReporting from '.';
+import * as VxTabulation from '../../tabulation';
+import {
+  BallotMeasureContest,
+  CandidateContest,
+  CountItemType,
+  PartyContest,
+  RetentionContest,
+} from '.';
+import {
+  isBallotMeasureContest,
+  isCandidateContest,
+  isRetentionContest,
+} from './types';
+
+type CandidateNameRecord = Record<CandidateId, Candidate['name']>;
+
+function assertGetDeepValue(root: unknown, valuePath: string) {
```

Comment:
This is used somewhat widely because many fields in the ERR schema are not technically required and therefore are optional. In many of these cases, the file would be useless without such fields. For example, a `CandidateSelection` has multiplicity `0..*` for the `Candidate` prop. Votes described by a `CandidateSelection` can't be mapped to a candidate without that prop, so we `assertGetDeepValue` rather than eg. use and handle a `Result`.